app/feature/insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/presentation/InsurancePresenter.kt

```diff
@@ -145,23 +68,96 @@ internal data class InsuranceUiState(
   }
 }
 
+internal class InsurancePresenter(
+  private val getInsuranceContractsUseCase: GetInsuranceContractsUseCase,
+  private val getCrossSellsUseCase: GetCrossSellsUseCase,
+  private val crossSellCardNotificationBadgeService: CrossSellCardNotificationBadgeService,
+) : MoleculePresenter<InsuranceScreenEvent, InsuranceUiState> {
+  @Composable
+  override fun MoleculePresenterScope<InsuranceScreenEvent>.present(
+    seed: InsuranceUiState,
+  ): InsuranceUiState {
+    var insuranceData by remember {
+      mutableStateOf<InsuranceData>(
+        InsuranceData(
```

Comment:
I would create a function here like `InsuranceData.fromUiState(seed)` in order to keep this present function as clean as possible.

Reply:
Done